app/content.ts

```diff
@@ -64,21 +64,32 @@ class Intercept {
     (document.head || document.documentElement).appendChild(sinon);
   };
   initScript = (request: requestObject) => {
-    let actualCode = `
+    let actualCode =`
     function remove(querySelector) {
       let elemToRemove = document.getElementById(querySelector);
       elemToRemove.parentNode.removeChild(elemToRemove);
     };
-    if (document.getElementById("tmpScript")) {
+    while(document.getElementById("tmpScript")) {
       remove("tmpScript");
     }
-    let request = ${JSON.stringify(request)};
-    let sinonServer = sinon.fakeServer.create();
-
-    sinonServer.respondWith('${request.method}', '${
-      request.url
-    }',[200, { "Content-Type": "application/json" },'[${JSON.stringify(request.responseText)}]']);
-    sinonServer.respondImmediately = true;`;
+    if (window.interceptor2) {
+      window.interceptor2.server.xhr.filters = [];
+    }
+    function sinonSingleHandler(request) {
+      this.server = sinon.fakeServer.create({ logger: console.log });
+      this.server.autoRespond = true;
+      this.server.xhr.useFilters = true;
+      // If the filter returns true, the request will not be faked - leave original
+      this.server.xhr.addFilter(function(method, url, async, username, password) {
+        const result = (request.url === url)
+        return !result
+      });
+      this.server.respondWith((xhr, id) => {
+        xhr.respond(200, { "Content-Type": "application/json" },'[${JSON.stringify(request.responseText)}]')

```

Comment:
Are we injecting this script for each recorded request.
If that is the case we are doing it wrong. 
once script will be injected for each time we say intercept. all the request selected will be handled by that.